src/kOS.Safe/Execution/Stack.cs

```diff
@@ -61,13 +60,14 @@ private object ProcessItem(object item)
             }
             if (item is double)
             {
-                var outOfBounds = Int32.MinValue < (double) item && (double) item < Int32.MaxValue;
+                var doubleItem = (double)item;
+                var outOfBounds = int.MinValue < doubleItem && doubleItem < int.MaxValue;
 
-                if (!Double.IsNaN((double)item) && outOfBounds)
+                if (!double.IsNaN(doubleItem) && outOfBounds)
                 {
-                    int intPart = Convert.ToInt32(item);
-                    if ((double)item == intPart)
-                        item = intPart;
+                    int intItem = Convert.ToInt32(item);
+                    if (doubleItem == intItem)
+                        item = intItem;
```

Comment:
This has nothing to do with your change, but I suspect this naming of variables was wrong all along.

I had a hard time following the logic here just now until I realized that the way it's written, `outOfBouds` is *actually* a boolean that means `isInBounds` - exactly the opposite of what it says.